operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/AbstractDependentResource.java

```diff
@@ -0,0 +1,31 @@
+package io.javaoperatorsdk.operator.api.reconciler.dependent;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+
+public abstract class AbstractDependentResource<R, P extends HasMetadata>
+    implements DependentResource<R, P> {
+
+  @Override
+  public void reconcile(P primary, Context context) {
+    var actual = getResource(primary);
+    var desired = desired(primary, context);
+    if (actual.isEmpty()) {
+      create(desired, primary, context);
+    } else {
+      if (!match(actual.get(), desired, context)) {
+        update(actual.get(), desired, primary, context);
+      }
```

Comment:
if match doesn't assert that it has been reconciled. The secondary R may exist prior to the primary one.  We should be able to put in the context what has been observed to be desired, so that the reconciler can update the status if it is empty.

In the case of mysql schema operator, if the database exists when the primary is created on the cluster, the status is never updated. But, it tricky here because the user is not symetric due to password.

if we split, schema and user in two different dependent. Thus, User::match cannot be sure actual = desired because the password may not be the desired one. So if the user exist prior to the CR, either it should fail the reconcile or the password should be changed and secret updated/recreated.

Reply:
> if match doesn't assert that it has been reconciled. The secondary R may exist prior to the primary one. We should be able to put in the context what has been observed to be desired, so that the reconciler can update the status if it is empty.

Not sure I'm following this. We read the R, and compare it to a target state, this is how P gets reconciled (at least this is a part of it). 

In MySQL there is no update now available for the password. But the whole mysql we will revisit shortly.